metadata-builders/src/main/java/com/linkedin/metadata/builders/search/DatasetIndexBuilder.java

```diff
@@ -12,24 +12,59 @@
 import com.linkedin.metadata.search.DatasetDocument;
 import com.linkedin.metadata.snapshot.DatasetSnapshot;
 import com.linkedin.schema.SchemaMetadata;
-import lombok.extern.slf4j.Slf4j;
-
-import javax.annotation.Nonnull;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import lombok.extern.slf4j.Slf4j;
 
 @Slf4j
 public class DatasetIndexBuilder extends BaseIndexBuilder<DatasetDocument> {
+  private static final Map<String, Character> PLATFORM_DELIMITER_MAP =
+      Collections.unmodifiableMap(new HashMap<String, Character>() {
```

Comment:
These are already defined here: https://github.com/linkedin/datahub/blob/master/gms/impl/src/main/resources/DataPlatformInfo.json and can be accessed via this endpoint: https://github.com/linkedin/datahub/blob/master/gms/impl/src/main/java/com/linkedin/metadata/resources/dataplatform/DataPlatforms.java

Reply:
Agreed.. !!   

This would require injecting restli client / remote dao into dataset index builder (which we don't do internally).    
+ @jplaisted.  This is our first use case in OSS where this injection is needed into Index builders.

Reply:
I don't think making a network call to access static information is a good idea.

The real issue here is that the index builders really should be part of the GMS (well, internally at least, part of that MP, rather than metadata-models).

Basically, this should be a common, static library that both the index builder and GMS can compile against. There's no reason for this to be a network call.

Reply:
What if dataplatform entity is hosted by one GMS and dataset entity is hosted different GMS. At that point, using a static file is an implementation detail of dataplatform GMS. 

Index Builders can have a client side caching instead of making network call for processing of every record.

Reply:
Perhaps this is out of scope for this PR but maybe DataPlatformInfo should not be static? 

For open source, there is a plethora of platforms out there that makes it infeasible to track in a single code file (AWS' S3 comes to mind). My outsider (and perhaps implementation-naive) perspective makes me think this should be a user PUT-able endpoint so that users have full flexibility to define any given platform.

I know our company would use this since we have proprietary platforms we'd like for DataHub to track.

Reply:
Agree. We should probably expose an API to CRUD the data platforms. Let's create a separate issue to track that, @nagarjunakanamarlapudi?

Reply:
Agree on enabling CRUD operations on DataPlatformInfo https://github.com/linkedin/datahub/issues/2018

Reply:
Today this information is static. Until we have the infrastructure for it to be dynamic, I disagree with adding that additional / unneeded moving piece.

Reply:
Ok, so spoke offline with Naga, and the proposal is to have this PR just do the network calls now. I think that is reasonable, however:

1) We need to delete `RegisteredIndexBuilders` and construct the builders in the job directly, so we can inject dependencies. We've already done this internally (we have a different set of jobs internally...).
2) There are many datasets, potentially. This information is very unlikely to change. If we want to make a network call to retrieve this information, we should cache it.

Reply:
And step 1 should ideally be a different PR that is submitted before this one.

Reply:
Created this issue for injecting restli client into index builders  https://github.com/linkedin/datahub/issues/2023